### ISSUE TITLE
Accessing static variables starting with an underscore in class functions emits W0212

### ISSUE THREAD

**Issue by xuhdev** [PyCQA/pylint#2933]
### Steps to reproduce
1. Have the following content saved as tmp.py:

```python
class MyClass:
    _static_member = 1

    def my_fun(self):
        print(__class__._static_member)
```

2. Run `pylint tmp.py`

### Current behavior

Output

```
test.py:1:0: C0111: Missing module docstring (missing-docstring)
test.py:1:0: C0111: Missing class docstring (missing-docstring)
test.py:4:4: C0111: Missing method docstring (missing-docstring)
test.py:5:14: W0212: Access to a protected member _static_member of a client class (protected-access)
test.py:4:4: R0201: Method could be a function (no-self-use)
test.py:1:0: R0903: Too few public methods (1/2) (too-few-public-methods)
```

Note the warning W0212.

### Expected behavior

W0212 shouldn't be in the output.

### pylint --version output

```
pylint 2.3.1
astroid 2.2.5
Python 3.7.3rc1 (default, Mar 13 2019, 11:01:15) 
[GCC 8.3.0]
```

**Comment by PCManticore** [comment]
This is a very contrived example. Why don't you simply use `self` instead of `__class__` to access the attribute?

**Comment by xuhdev** [comment]
Because we are accessing static class members. This has two benefits: Better readability (explicitly static) and sometimes the code makes a difference. For example, in the following example, `my_fun` prints two different lines.

```python
class MyClass:
    _static_member = 1

    def my_fun(self):
        self._static_member = 10
        print(__class__._static_member)
        print(self._static_member)
```
